libsolidity/formal/CHC.cpp

```diff
@@ -196,8 +196,12 @@ void CHC::endVisit(ContractDefinition const& _contract)
 	connectBlocks(m_currentBlock, summary(_contract));
 
 	setCurrentBlock(*m_constructorSummaries.at(&_contract));
-	m_queryPlaceholders[&_contract].push_back({smtutil::Expression(true), errorFlag().currentValue(), m_currentBlock});
-	connectBlocks(m_currentBlock, interface(), errorFlag().currentValue() == 0);
+	auto constructor = _contract.constructor();
+	auto txConstraints = state().txConstraints();
+	if (!constructor || !constructor->isPayable())
+		txConstraints = txConstraints && state().txMember("msg.value") == 0;
```

Comment:
If contract contains no constructor or contains a non-payable constructor, msg value must be zero. Why is this constraint added during end visit of contract though?